common.eai

```diff
@@ -10455,7 +10455,7 @@ function StartUnitAM takes integer ask_qty, integer unitid, integer town, intege
     endif
 
   // Intentionally not using have_qty as the build lumber mill at base can not be picked up by the count as it builds somewhere else
-    if unitid == racial_lumber and GetUnitCount(old_id[unitid]) < 1 and town <= 0 then
+    if unitid == racial_lumber and ((GetUnitCount(old_id[unitid]) < 1 and town < 1) or (GetUnitCount(old_id[unitid]) < TownCount(racial_expansion) and town > 0)) then
```

Comment:
This is incorrect logic and not sure what you trying to do.

Intent is to only build lumber mill near trees at the main base. But your change will make it build more than 1 at the main base even if the desired town is not the main base.

Reply:
Previously, have some images is AI will build very more graveyard
SO . At first, my consideration was that a game only needed to have one `racial_lumber` present at the same time

But I saw that you added `and town <= 0`
I thought you didn't limit the expansion build lumber , yes , it seems unnecessary to limit . For example, after collecting the wood from the main base, you can find a new place to collect it -- This looks really fine

But in this way, when `town > 0` ,  it is still impossible to limit the quantity . 
So I added a constraint to handle the infinite construction of extensions

Reply:
If town > 0 we don't want to limit the quantity. It can build those at expansions as normal if it has been requested for more than 1, standard build logic will handle any additional lumber mills in those cases. 

This function is only to catch the very first lumber mill and build it closer to the trees. It shouldn't be catching any others.

Reply:
If there were no restrictions, you might really see a cemetery somewhere

Reply:
Yes, if a strategy requests that it should do so.